dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/common/SqlQueryExecutor.java

```diff
@@ -54,27 +62,55 @@ public SqlQueryExecutor(@Qualifier("analyticsReadOnlyJdbcTemplate") JdbcTemplate
    * @throws IllegalArgumentException if the query argument is null.
    */
   @Override
-  public SqlQueryResult find(SqlQuery query) {
-    notNull(query, "The 'query' must not be null");
+  public SqlQueryResult find(@Nonnull SqlQueryCreator queryCreator) {
```

Comment:
We should find a way here. We should aim to keep the `SqlQuery` to be executed.
Can't we pass in the `List<DimensionIdentifier<DimensionParam>>`, as a parameter along with the `Query`.
If not, let's find another way, to avoid passing a `SqlQueryCreator` as argument.

Reply:
I will look into it